src/python/pants/backend/nfpm/config.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from typing import TypedDict
+
+import yaml
+
+from pants.backend.nfpm.fields.contents import (
+    NfpmContentFileGroupField,
+    NfpmContentFileModeField,
+    NfpmContentFileMtimeField,
+    NfpmContentFileOwnerField,
+)
+from pants.engine.target import Target
+
+
+class OctalInt(int):
+    # noinspection PyUnusedLocal
+    @staticmethod
+    def represent_octal(dumper: yaml.representer.BaseRepresenter, data: int) -> yaml.Node:
+        # YAML 1.2 octal: 0o7777 (py: f"0o{data:o}" or f"{data:#o}" or oct(data))
+        # YAML 1.1 octal: 07777 (py: f"0{data:o}")
+        # Both octal reprs are supported by `gopkg.in/yaml.v3` which parses YAML in nFPM.
+        # See: https://github.com/go-yaml/yaml/tree/v3.0.1#compatibility
+        # PyYAML only supports reading YAML 1.1, so we use that.
+        return yaml.ScalarNode("tag:yaml.org,2002:int", f"0{data:o}")
+
+
+# This is an unfortunate import-time side effect: PyYAML does registration globally.
+yaml.add_representer(OctalInt, OctalInt.represent_octal, Dumper=yaml.SafeDumper)
+
+
+class NfpmFileInfo(TypedDict, total=False):
+    # nFPM allows these to be None or missing.
+    # Each of the fields has a default, so in practice, these won't be None.
+    owner: str | None
+    group: str | None
+    mode: OctalInt | None
+    mtime: str | None
+
+
+def file_info(
+    target: Target, default_is_executable: bool | None = None, default_mtime: str | None = None
+) -> NfpmFileInfo:
+    mode = target[NfpmContentFileModeField].value
```

Comment:
Why not use a `FieldSet` to extract these fields?

Reply:
I don't think that makes sense, as I would have to construct a new field set just for this function.
This is where that function is used while iterating over transitive dependencies (`NfpmContent` is a `TypedDict` that represents a structure in the `nfpm.yaml` config file):
https://github.com/pantsbuild/pants/blob/5d0ce9c4a3392e91674839283ed9868499f083da/src/python/pants/backend/nfpm/util_rules/generate_config.py#L91-L109

Reply:
I suppose the call sites could move this function onto an `NfpmFileInfoFieldSet`. Then the call sites would end up like this:

```python
 for tgt in transitive_targets.dependencies: 
	     if tgt.has_field(NfpmContentDirDstField):  # an NfpmContentDir 
	         contents.append( 
	             NfpmContent( 
	                 type="dir", 
	                 dst=tgt[NfpmContentDirDstField].value, 
	                 file_info=NfpmFileInfoFieldSet.create(tgt).file_info(default_mtime), 
	             ) 
	         ) 
```

Is that better?

Reply:
OK. I moved this logic into some new field sets. I guess that does feel nicer. Thanks!